Assets/Scripts/Puzzels/Crane/Joystick.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using LekAanDek.Events;
+
+namespace LekAanDek.CranePuzzle
+{
+    /// <summary>
+    /// This class handles the joystick logic to move the crane left and right
+    /// </summary>
+    public class Joystick : MonoBehaviour
+    {
+
+        [SerializeField]
+        private Transform _joyStickTop;
+
+        private float _horizontalTilt = 0f;
+        private float _yRot = 0f;
+
+        [SerializeField]
+        private FloatEvent _craneRight;
+        [SerializeField]
+        private FloatEvent _craneLeft;
+
+        // Update is called once per frame
+        void Update()
+        {
+            _horizontalTilt = this.transform.rotation.eulerAngles.x;
+            _yRot = Mathf.Round(this.transform.rotation.eulerAngles.y);
+
+            // IF joystick is facing right
+            if (_horizontalTilt >270 && _horizontalTilt < 359 && _yRot == 270)
+            {
+                _horizontalTilt = Mathf.Abs(_horizontalTilt - 360);
+                // Turn crane left
+                _craneLeft.Raise(_horizontalTilt);
+            }
+            // IF joystick is facing left
+            else if (_horizontalTilt > 270 && _horizontalTilt < 359 && _yRot == 90)
```

Comment:
Try not to use magic numbers and create constants or variables out of these for clarity.